File: Javascript/longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs){
    if(!strs){
        return "";
    }

    var prefix = strs[0];

    for(let i = 0; i< strs.length; i++){
        while(strs[i].slice(0, prefix.length) != prefix){
            prefix = prefix.slice(0, -1);
            if(!prefix){
                return "";
            }
        }
    }
    return prefix;

};

File: Python3/longest-common-prefix.py
from typing import List

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) < 1:
            return ""
        
        prefix = strs[0]
        for string in strs:
            
            while string[:len(prefix)] != prefix:
                prefix = prefix[:-1]
                if not prefix:
                    return ""
                
        return prefix